subprojects/ide-native/src/integTest/groovy/org/gradle/ide/xcode/XcodeCppExternalSourceDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.ide.xcode
+
+import org.gradle.ide.xcode.fixtures.AbstractXcodeIntegrationSpec
+import org.gradle.integtests.fixtures.build.BuildTestFile
+import org.gradle.internal.hash.HashUtil
+import org.gradle.nativeplatform.fixtures.app.CppAppWithLibrary
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.GFileUtils
+import org.gradle.vcs.fixtures.GitRepository
+import org.junit.Rule
+
+import static org.gradle.ide.xcode.internal.XcodeUtils.toSpaceSeparatedList
+
+class XcodeCppExternalSourceDependenciesIntegrationTest extends AbstractXcodeIntegrationSpec {
```

Comment:
Should add some test coverage where the root project is only a container that does not define any components, and there are 2 subprojects with an application defined in one and library in the other.

Should also cover the case where there are sub projects that don't define any components.